set_datatype.hs

```diff
@@ -65,10 +65,6 @@ singleton x = Branch Leaf x Leaf
 -}
 insert :: Ord a => a -> Set a -> Set a
 insert n Leaf = Branch Leaf n Leaf
-insert n (Branch Leaf x Leaf) =
```

Comment:
This case is identical to the `Branch lb x rb` case, so there's no reason to keep it.